baybe/utils/dataframe.py

```diff
@@ -361,6 +361,50 @@ def df_uncorrelated_features(
     return data
 
 
+def df_add_noise_to_degenerate_rows(
+    df: pd.DataFrame, noise_level: float = 0.001
+) -> pd.DataFrame:
+    """Add noise to degenerate rows to being able to numerically distinguish them.
+
+    Note that the dataframe is changed in-place and also returned.
+
+    Args:
+        df: The dataframe to be modified.
+        noise_level: The magnitude of noise relative to the min-max range of
+            values for each column.
```

Comment:
"Level" is perhaps a bit misleading. I guess you want exactly the same meaning as we have `simulate_experiment`, where we call it `noise_percent`, which makes the "relative" nature more explicit. That said, what I don't like about the latter is that it is actually a float, not a percentage :D So unclear what is the best solution, ideally we should target a naming that is consistent if they really represent the same idea (we could also rename it in the simulation module in the next iteration). Ideas

Reply:
happy to call it `percentage` but then we should expect it in range 0-100 and not 0-1

Reply:
yet another short alternative would be `noise_ratio`, which sort of also complies with the common term "signal-to-noise-ratio".  I'm open to everything and let you decide – you can even keep it the way it is if you think it's the best option. To me, it is more important that we try to harmonize this somehow across different functions and use a name that best reflects its meaning

Reply:
ok changed to `noise_ratio`

Reply:
Great thanks. I'll then probably also rename this in the simulation module once there is an opportunity for it